go/src/koding/klient/machine/client/kite.go

```diff
@@ -46,10 +46,6 @@ func (kb *KiteBuilder) Ping(dynAddr DynamicAddrFunc) (machine.Status, machine.Ad
 // Build builds new kite client that will connect to machine using provided
 // address.
 func (kb *KiteBuilder) Build(ctx context.Context, addr machine.Addr) Client {
-	if _, err := kb.pool.Get(addr.Value); err != nil {
-		return NewDisconnected(ctx)
-	}
-
```

Comment:
Indeed it is, but it's also duplicated by `Ping` [method](https://github.com/koding/koding/blob/7ff210291a6f0a7358c8470d709446119a13876b/go/src/koding/klient/machine/client/kite.go#L36). With this one it would be harder to recover. Also, `Build` is done only once per valid klient.